src/api/config.c

```diff
@@ -612,6 +612,49 @@ m64p_error ConfigShutdown(void)
 /* Selector functions, exported outside of the Core */
 /* ------------------------------------------------ */
 
+EXPORT m64p_error CALL ConfigExternalGetParameter(const char *FileName, const char *SectionName, const char *ParamName, char* ParamPtr)
+{
+    FILE *fPtr;
+    char buffer[256];
+    int foundSection = 0;
+    if (ParamPtr == NULL || SectionName == NULL || ParamName == NULL)
+        return M64ERR_INPUT_INVALID;
+    if (FileName == NULL || (fPtr = fopen(FileName, "rb")) == NULL)
+    {
+        DebugMessage(M64MSG_ERROR, "Unable to open config file '%s'.", FileName);
+        return M64ERR_INPUT_INVALID;
+    }
+    while (fgets(buffer, 255, fPtr) != NULL)
+    {
+        char *line = buffer;
+        ini_line l = ini_parse_line(&line);
+        switch (l.type)
+        {
+            case INI_SECTION:
+                if (osal_insensitive_strcmp(SectionName, l.name) == 0)
+                    foundSection = 1;
+                else
+                    foundSection = 0;
+                break;
+            case INI_PROPERTY:
+                if (foundSection)
+                {
+                    if (osal_insensitive_strcmp(ParamName, l.name) == 0)
+                    {
+                        fclose(fPtr);
+                        strncpy(ParamPtr, l.value, 256);
```

Comment:
haha I didn't realize there was a hierarchy :P 

To me it makes more sense to pass the function a variable I allocated. It seems strange to me as a plugin dev to free a pointer that I never malloc'ed.

Generally I think it's fair to assume that a dev should read the docs if they are going to implement an API. I think passing a size variable would be fine, but then you are passing a lot of variable into this function, and you are just protecting against a situation where someone didn't read the docs.